core/src/main/java/org/bitcoinj/core/BitcoinSerializer.java

```diff
@@ -288,7 +288,9 @@ private Message makeMessage(String command, int length, byte[] payloadBytes, byt
         } else if(command.equals("qsendrecsigs")) {
             return new QuorumSendRecoveredSignatures(params);
         } else if(command.equals("islock")) {
-            return new InstantSendLock(params, payloadBytes);
+            return new InstantSendLock(params, payloadBytes, InstantSendLock.ISLOCK_VERSION);
+        } else if(command.equals("isdlock")) {
+            return new InstantSendLock(params, payloadBytes, InstantSendLock.ISDLOCK_VERSION);
```

Comment:
This is the part that will handle received deterministic instant send messages, which are "isdlock".